capplet/gsm-properties-dialog.c

```diff
@@ -583,28 +572,25 @@ setup_dialog (GsmPropertiesDialog *dialog)
                                                  G_TYPE_STRING,
                                                  G_TYPE_OBJECT,
                                                  G_TYPE_STRING);
-        tree_filter = gtk_tree_model_filter_new (GTK_TREE_MODEL (dialog->list_store),
-                                                 NULL);
-        g_object_unref (dialog->list_store);
-        dialog->tree_filter = tree_filter;
 
-        gtk_tree_model_filter_set_visible_column (GTK_TREE_MODEL_FILTER (tree_filter),
+        dialog->tree_filter = gtk_tree_model_filter_new (GTK_TREE_MODEL (dialog->list_store),
+                                                         NULL);
+
```

Comment:
Is this missing?
```
g_object_unref (dialog->list_store);
```

Reply:
No it isn't. It is not unreferenced in the code below:

https://gitlab.gnome.org/GNOME/gtk/-/blob/master/testsuite/gtk/filtermodel.c#L125-154
```C
static void
filter_test_setup_generic (FilterTest    *fixture,
                           gconstpointer  test_data,
                           int            depth,
                           gboolean	  empty,
                           gboolean       unfiltered)
{
  const GtkTreePath *vroot = test_data;
  GtkTreeModel *filter;

  fixture->store = create_tree_store (depth, !empty);

  g_signal_connect_swapped (fixture->store, "row-changed",
                            G_CALLBACK (filter_test_store_signal), fixture);

  /* Please forgive me for casting const away. */
  filter = gtk_tree_model_filter_new (GTK_TREE_MODEL (fixture->store),
                                      (GtkTreePath *)vroot);
  fixture->filter = GTK_TREE_MODEL_FILTER (filter);

  if (!unfiltered)
    gtk_tree_model_filter_set_visible_column (fixture->filter, 1);

  /* We need a tree view that's listening to get ref counting from that
   * side.
   */
  fixture->tree_view = gtk_tree_view_new_with_model (filter);

  fixture->monitor = signal_monitor_new (filter);
}
```